### PR TITLE
850-support-email-participants-in-message-store

### LINKED ISSUE #850: Support email participants in message store

### PR DESCRIPTION
**Pull request by gabriel0003**

### CODE REVIEW

File: packages/server/message-store-api/service/MessageStoreService.go [hunk 1]
```diff
@@ -226,26 +285,20 @@ func (s *MessageService) getMessagePreview(input *msProto.InputMessage) string {
...
+	if err != nil {
+		contact, err := s.customerOSService.GetContactWithEmailOrCreate(tenant, initiatorIdentifier)
+		if err != nil {
 			return nil, err
-		} else {
-			initiatorId = &user.Id
 		}
+		return &Participant{Id: contact.Id, Type: entity.CONTACT}, nil
+	} else {
+		return &Participant{Id: user.Id, Type: entity.CONTACT}, nil
```
Comment (tsearle) [inline packages/server/message-store-api/service/MessageStoreService.go]: I think this should be entity.USER

File: packages/server/message-store-api/service/MessageStoreService.go [hunk 2]
```diff
@@ -226,26 +285,20 @@ func (s *MessageService) getMessagePreview(input *msProto.InputMessage) string {
...
 	}
-
-	return initiatorId, nil
 }
 
-func (s *MessageService) saveConversationEvent(tenant string, conversation *Conversation, input *msProto.InputMessage, initiatorId string, senderType entity.SenderType) entity.ConversationEvent {
+func (s *MessageService) saveConversationEvent(tenant string, conversation *Conversation, input *msProto.InputMessage, participant *Participant) entity.ConversationEvent {
```
Comment (tsearle) [inline packages/server/message-store-api/service/MessageStoreService.go]: variable name is confusing, it should remain called sender despite being of type Participant, as it is being used to indicate which participant sent the message

File: packages/server/message-store-api/service/MessageStoreService.go [hunk 3]
```diff
@@ -203,15 +217,60 @@ func (s *MessageService) SaveMessage(ctx context.Context, input *msProto.InputMe
...
+	var lastSenderType = ""
+	if initiator.Type == entity.CONTACT {
+		lastSenderType = "CONTACT"
+	} else if initiator.Type == entity.USER {
+		lastSenderType = "USER"
+	}
+	return lastSenderType
+}
+
+func (s *MessageService) getThreadIdAndParticipantsFromMail(tenant string, threadId *string, participants []Participant, entityType entity.EventType, input *msProto.InputMessage) error {
```
Comment (tsearle) [inline packages/server/message-store-api/service/MessageStoreService.go]: this won't work, passing an array creates a copy of the array,
and append creates a new array...

File: packages/server/message-store-api/service/MessageStoreService.go (hunk 1 above)
Comment (gabriel0003) [reply packages/server/message-store-api/service/MessageStoreService.go]: Indeed

File: packages/server/message-store-api/service/MessageStoreService.go (hunk 2 above)
Comment (gabriel0003) [reply packages/server/message-store-api/service/MessageStoreService.go]: Fixed

File: packages/server/message-store-api/service/MessageStoreService.go (hunk 3 above)
Comment (gabriel0003) [reply packages/server/message-store-api/service/MessageStoreService.go]: Fixed

**Review by tsearle — APPROVED** [review]

**Pull request merged**